tests/solver.rs

```diff
@@ -1429,6 +1432,203 @@ fn test_explicit_root_requirements() {
     "###);
 }
 
+#[test]
+#[traced_test]
+fn test_conditional_requirements() {
+    let mut provider = BundleBoxProvider::new();
+
+    // Add packages
+    provider.add_package("a", 1.into(), &["b"], &[]); // a depends on b
+    provider.add_package("b", 1.into(), &[], &[]); // Simple package b
+    provider.add_package("c", 1.into(), &[], &[]); // Simple package c
+
+    // Create conditional requirement: if b=1 is installed, require c
+    let b_spec = Spec::parse_union("b 1").next().unwrap().unwrap();
+    let c_spec = Spec::parse_union("c 1").next().unwrap().unwrap();
+
+    let b_version_set = provider.intern_version_set(&b_spec);
+    let c_version_set = provider.intern_version_set(&c_spec);
+
+    let conditional_req = ConditionalRequirement::new(b_version_set, c_version_set.into());
```

Comment:
Ideally you add a little string conversion for parsing this so you can do e.g. `c 1; if b 1..2` that would make the tests more readable.